model/event/event_finder.go

```diff
@@ -161,29 +170,41 @@ func TaskEventsInOrder(id string) db.Q {
 }
 
 // Distro Events
-// DistroPrimaryEventsForId returns the non-AMI specific events for the distro.
-func DistroPrimaryEventsForId(id string) db.Q {
-	filter := ResourceTypeKeyIs(ResourceTypeDistro)
-	filter[ResourceIdKey] = id
-	filter[TypeKey] = bson.M{"$ne": EventDistroAMIModfied}
 
-	return db.Query(filter)
+// FindLatestPrimaryDistroEvents return the most recent non-AMI events for the distro.
+func FindLatestPrimaryDistroEvents(id string, n int) ([]EventLogEntry, error) {
+	return Aggregate(latestDistroEventsPipeline(id, n, false))
 }
 
-// DistroAMIModifiedForId returns the query for AMI modified distro events.
-func DistroAMIModifiedForId(id string) db.Q {
-	filter := ResourceTypeKeyIs(ResourceTypeDistro)
-	filter[ResourceIdKey] = id
-	filter[TypeKey] = EventDistroAMIModfied
-	return db.Query(filter).Sort([]string{"-" + TimestampKey})
-}
-
-func MostRecentDistroEvents(id string, n int) db.Q {
-	return DistroPrimaryEventsForId(id).Sort([]string{"-" + TimestampKey}).Limit(n)
+// FindLatestAMIModifiedDistroEvent returns the most recent AMI event. Returns an empty struct if nothing exists.
+func FindLatestAMIModifiedDistroEvent(id string) (EventLogEntry, error) {
```

Comment:
nit: this could just return a pointer if we don't want to return empty structs. I don't care that much, it just makes it consistent with other `FindOne`-style queries.

Reply:
I'm kind of okay with this since then we don't need to make a nil check for latest timestamp.